### PR TITLE
Disable an item from being dragged over a stamped item.

### PR DESCRIPTION
**Pull request by drewpc**
When an item is dragged onto a stamped item, this code moves the dragged item back to its original position and calls "layout" again.  This was tested with jQuery UI but not Draggabilly.

### CODE REVIEW

**Comment by druellan** [comment]
Thanks @drewpc. I used your code to build my own function, but creating a new item property `locked` (to avoid messing with `stamp`). This is my fork: https://github.com/druellan/packery

**Comment by drewpc** [comment]
I updated my code address the issues mentioned, update with the latest version of Packery, included unit tests, and added a test case on CodePen (http://codepen.io/anon/pen/vOvGEV).  In order to use the Rect.overlaps() logic, I needed to ensure that Rect.overlaps() tests for <= and >= instead of < and >.  What do you think of that?  Would that break existing Packery implementations?

**Comment by drewpc** [comment]
After further testing, I removed the Rect.overlaps() changes that I made/proposed earlier.

**Comment by drewpc** [comment]
I just updated/merged changes from your library so this can be automatically merged if you approve.  Sorry for the weird set of comments--I tried to rebase my fork and it screwed things up.

File: js/packery.js [hunk 1]
```diff
@@ -429,6 +429,33 @@ Packery.prototype.itemDragEnd = function( elem ) {
...
   if ( item.needsPositioning ) {
+    // Check to see if the item being positioned is overlapping a stamped item.
+    for(var i = 0; i < this.stamps.length; i++) {
+      if(this.stamps[i] == item.element) {
+          // The currently moving item is stamped as well, so just ignore that one.
+          continue;
+      }
+      // Note: the stamped item MUST be an actual "item" so we can get the rect.x,rect.y coordinates.
+      // Otherwise the stamped item is just a DOM element.
+      var stamped = this.getItem(this.stamps[i]);
```
Comment (desandro) [inline js/packery.js]: You may want to look at using [`_getElementOffset`](https://github.com/metafizzy/outlayer/blob/v1.4.2/outlayer.js#L583) and [`_setRectSize`](https://github.com/metafizzy/packery/blob/v1.4.3/js/packery.js#L185), similar to how its used in [`_manageStamp`](https://github.com/metafizzy/packery/blob/v1.4.3/js/packery.js#L238)

**Comment by desandro** [comment]
@drewpc Thanks so much for this contribution. Great to see you have found a solution. While this may work, I'm not ready to merge it in. I'd like to consider the benefit of this feature and if its worth adding more complexity to Packery.

**Comment by kdamken** [comment]
@desandro This issue may prevent us from using this commercially - any news on whether this or another solution had been implemented?

**Comment by desandro** [comment]
This PR was useful for Packery v1. But now that Packery v2 has been released, it can no longer be used. As such, I'm closing this PR. Thanks again for this contribution.

@kdamken Packery v2 does have improvements for its drag and drop behavior, but still may experience this issue.

**Comment by drewpc** [comment]
We haven't moved to Packery v2 yet, so I haven't updated the PR to reflect those changes.  If I update the PR can we reopen this?  Or should I just submit another one?

**Comment by desandro** [comment]
Either works. I'd prefer for the PR to show only code changes. Right now its showing all the updated code since the original PR is so old.

**Pull request closed without merging**